torch/_inductor/lowering.py

```diff
@@ -734,7 +734,10 @@ def expand(x, sizes):
         return x
 
     x_size_product = V.graph.sizevars.size_hint(sympy_product(x.get_size()))
-    if x_size_product > 0:
+    if (
+        not any(V.graph.sizevars.shape_env.is_unbacked_symint(s) for s in sizes)
+        and x_size_product > 0
```

Comment:
To be clear, is it the `x_size_product > 0` that is failing? The condition here seems flipped; if I can't tell if `x_size_product` is non-zero or not, it seems better to assume that it is non-zero, because that's more common.

Reply:
Found that we no longer need this change after switching to use ExternKernelAlloc